### DISCUSSION TITLE
Support `ATTACH` property

### DISCUSSION THREAD

**Discussion by Gummibeer** [spatie/icalendar-generator#56] [Ideas]
RFC 5545 has a definition for `ATTACH` property which allows adding attachments to a single event. The attachment should be added as an URL to the file - but in theory also supports binary embedding.
The request would be to allow adding attachments via URL.

https://icalendar.org/iCalendar-RFC-5545/3-8-1-1-attachment.html

**PR:** https://github.com/spatie/icalendar-generator/pull/57

**Comment by rubenvanassche** [comment]
Hi Tom,

Started working on an implementation: https://github.com/spatie/icalendar-generator/pull/57/files#diff-23b65a7af5b48c1bf5916d6ba27b53533227ffde23424083e5fd775979a322cb . But that led me to completely rethinking how this package should be tested. This is how the test looks like: https://github.com/spatie/icalendar-generator/pull/57/files#diff-8a56c15fe657efe53ee1bc6dc39cb3bcf4c8477401fd2135717631f08f904bfcR481-R505.

Two questions for you:
- What do you think about the `attachment` method? Does it fulfill your requirement?
- What do you think about the expectation test draft PR I've created?

**Reply by Gummibeer** [reply]
Following the spec I think that the `FMTTYPE` is missing. 🤔
```ics
ATTACH;FMTTYPE=application/postscript:ftp://example.com/pub/reports/r-960812.ps
```
And because of that I would prefer a `attach()` method doing `$this->attachments[] = ...` so that I can attach document per document.
Regarding testing: I think that the expectations are a good way but I would also use a lot of snapshot testing as it also allows to import the file in any calendar app to check that the result is the expected behavior.

**Reply by rubenvanassche** [reply]
Wel it's actually an optional argument when you're working with non-binary data, but yeah, maybe it's a better idea to include it.

**Reply by rubenvanassche** [reply]
Version 2.3.0 with attachments is now released!
